cbits/posix/find_executable.c

```diff
@@ -34,13 +64,13 @@ static char *find_in_search_path(char *search_path, const char *filename) {
 #pragma GCC diagnostic push
 #pragma GCC diagnostic ignored "-Wmaybe-uninitialized"
 #endif
-        const int tmp_len = filename_len + 1 + strlen(path) + 1;
+        const int tmp_len = workdir_len + 1 + filename_len + 1 + strlen(path) + 1;
 #if defined(__GNUC__) && __GNUC__ == 6 && __GNUC_MINOR__ == 3
 #pragma GCC diagnostic pop
 #endif
         char *tmp = malloc(tmp_len);
-        snprintf(tmp, tmp_len, "%s/%s", path, filename);
-        if (is_executable(tmp)) {
+        snprintf(tmp, tmp_len, "%s/%s/%s", working_dir, path, filename);
```

Comment:
Does this behave correctly in the case that `path` is an absolute path?

Reply:
No, I believe it does not. Good catch!